app/public/cantusdata/helpers/mei_processing/mei_validation.py

```
@@ -0,0 +1,54 @@
+    """
+    # MEIParser takes a path, so the document has to reach the filesystem to be
+    # checked. The temporary directory is removed before returning either way.
+    tmp_dir = mkdtemp(prefix="mei-validate-")
+    try:
+        tmp_path = path.join(tmp_dir, "submission.mei")
+        with open(tmp_path, "w", encoding="utf-8") as tmp_file:
+            tmp_file.write(mei)
+        try:
+            parser = MEIParser(tmp_path)
+        except XMLSyntaxError as exc:
+            raise MEIValidationError(f"Not well-formed XML: {exc}") from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)